content_scripts/mode_normal.js

```diff
@@ -213,6 +213,18 @@ const NormalModeCommands = {
     }
   },
 
+  findSelected() {
+    let selection = window.getSelection().toString();
+    if (!selection) return;
+    return FindMode.findNext(false, selection);
```

Comment:
If we do want to avoid passing in a query (see comments above), then we can add the two lines above this return.
```
    FindMode.updateQuery(selection);
    FindMode.saveQuery();
```
As you found, to fix the undefined object error, we also need to add an initialization to the start or the `updateQuery` method, like:
`this.query = {};`
Then everything works without needing to pass a query into the `findNext` or `getQuery` methods.

Reply:
You're right, it is a better approach. Thanks!